Tasks/CargoAuthenticateV0/cargoauthenticatemain.ts

```diff
@@ -0,0 +1,76 @@
+import * as path from 'path';
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as toml from 'toml';
+import * as fs from 'fs';
+import * as constants from './constants';
+import * as util from 'azure-pipelines-tasks-packaging-common/util';
+import * as pkgLocationUtils from 'azure-pipelines-tasks-packaging-common/locationUtilities';
+import * as url from 'url';
+
+async function main(): Promise<void> {
+    tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+    try {
+        let configtoml = tl.getInput(constants.CargoAuthenticateTaskInput.WorkingFile);
+        if (!tl.exist(configtoml)) {
+            throw new Error(tl.loc('ConfigTomlDoesNotExist', configtoml));
+        }
+        else {
+            console.log(tl.loc("AuthenticatingThisConfigToml", configtoml));
+        }
+        
+        // These two formats will work
+        // [registries]
+        // zhentan-test = { index = "sparse+https://pkgs.dev.azure.com/codesharing-SU0/zhentan-test/_packaging/zhentan-test/Cargo/index/" }
+        // [registries.zhentan-test1]
+        // index = "sparse+https://pkgs.dev.azure.com/codesharing-SU0/zhentan-test/_packaging/zhentan-test/Cargo/index/"
+        let configtomlFile = fs.readFileSync(configtoml, 'utf8');
+        var result = toml.parse(configtomlFile);
+        if (!result.registries)
+        {
+            throw new Error("config.toml must contains registries field");
+        }
+
+        let packagingLocation: pkgLocationUtils.PackagingLocation;
+        try {
+            packagingLocation = await pkgLocationUtils.getPackagingUris(pkgLocationUtils.ProtocolType.Npm);
```

Comment:
Should this be cargo, I don’t think it matters cos we only care about the host, but be consistent for someone else who might want to make changes to the code

Reply:
Yes, but if we add Cargo to the common packaging code, we will need to wait for it to be deployed before it's available for usage here. I will make that change in this PR, and update this to Cargo once it's deployed to npmjs.